Can't build in windows from v0.7.0

Log:

```powershell
   Compiling cargo_metadata v0.19.1
   Compiling cargo-selector v0.7.0
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> C:\Users\dzdzb\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cargo-selector-0.7.0\src\main.rs:8:9
  |
8 |     os::unix::process::ExitStatusExt as _,
  |         ^^^^ could not find `unix` in `os`
  |
note: found an item that was configured out
 --> /rustc/9fc6b43126469e3858e2fe86cafb4f0fd5068869\library\std\src\os\mod.rs:36:9
note: the item is gated here
 --> /rustc/9fc6b43126469e3858e2fe86cafb4f0fd5068869\library\std\src\os\mod.rs:28:1
note: found an item that was configured out
 --> /rustc/9fc6b43126469e3858e2fe86cafb4f0fd5068869\library\std\src\os\mod.rs:85:9
note: the item is gated here
 --> /rustc/9fc6b43126469e3858e2fe86cafb4f0fd5068869\library\std\src\os\mod.rs:84:1

error[E0599]: no method named `signal` found for struct `ExitStatus` in the current scope
   --> C:\Users\dzdzb\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cargo-selector-0.7.0\src\main.rs:155:49
    |
155 |             } else if let Some(signal) = status.signal() {
    |                                                 ^^^^^^ method not found in `ExitStatus`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `cargo-selector` (bin "cargo-selector") due to 2 previous errors
error: failed to compile `cargo-selector v0.7.0`, intermediate artifacts can be found at `C:\Users\dzdzb\AppData\Local\Temp\cargo-installrDg3tZ`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

but, can install v0.6.0.
Since https://github.com/lusingander/cargo-selector/commit/748c68df9e313d2423255897681fc6821ff2bf52, I can no longer build it, so I think that's related to it.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,7 +18,11 @@ jobs:
         versions:
           - stable
           - '1.79.0' # MSRV
-    runs-on: ubuntu-latest
+        os:
+          - ubuntu-24.04
+          - macos-14
+          - windows-2022
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master

--- a/src/main.rs
+++ b/src/main.rs
@@ -5,9 +5,8 @@ mod util;
 
 use std::{
     io::{stderr, BufWriter, Stderr},
-    os::unix::process::ExitStatusExt as _,
     panic,
-    process::ExitCode,
+    process::{ExitCode, ExitStatus},
 };
 
 use clap::{Args, Parser, ValueEnum};
@@ -24,6 +23,9 @@ use crate::{
     matcher::Matcher,
     tui::{Ret, Tui},
 };
+
+#[cfg(unix)]
+use std::os::unix::process::ExitStatusExt as _;
 
 #[derive(Debug, Parser)]
 #[command(name = "cargo", bin_name = "cargo")]
@@ -121,6 +123,26 @@ fn initialize_panic_handler(inline: bool) {
     }));
 }
 
+#[cfg(unix)]
+fn to_exit_code(status: ExitStatus) -> ExitCode {
+    if let Some(code) = status.code() {
+        ExitCode::from(code as u8)
+    } else if let Some(signal) = status.signal() {
+        ExitCode::from(signal as u8 + 128)
+    } else {
+        ExitCode::FAILURE
+    }
+}
+
+#[cfg(windows)]
+fn to_exit_code(status: ExitStatus) -> ExitCode {
+    if let Some(code) = status.code() {
+        ExitCode::from(code as u8)
+    } else {
+        ExitCode::FAILURE
+    }
+}
+
 fn main() -> std::io::Result<ExitCode> {
     let Cli::Selector(args) = Cli::parse();
     let SelectorArgs {
@@ -150,13 +172,7 @@ fn main() -> std::io::Result<ExitCode> {
         Ret::Quit => ExitCode::SUCCESS,
         Ret::Selected(t, a) => {
             let status = cargo::exec_cargo_run(&t, &a);
-            if let Some(code) = status.code() {
-                ExitCode::from(code as u8)
-            } else if let Some(signal) = status.signal() {
-                ExitCode::from(signal as u8 + 128)
-            } else {
-                ExitCode::FAILURE
-            }
+            to_exit_code(status)
         }
         Ret::NotSelected => {
             eprintln!("no command selected");

